--- a/main.c
+++ b/main.c
@@ -0,8 +0,7 @@ static void add_default_include_paths(char *argv0) {
-  // We expect that chibicc-specific include files are installed
-  // to ./include relative to argv[0].
-  strarray_push(&include_paths, format("%s/include", dirname(strdup(argv0))));
-
-  // Add standard include paths.
-  strarray_push(&include_paths, "/usr/local/include");
-  strarray_push(&include_paths, "/usr/include/x86_64-linux-gnu");
-  strarray_push(&include_paths, "/usr/include");
+  strarray_push(&include_paths, CHIBICC_ROOT "/include");
+  char *paths = strdup(getenv("C_INCLUDE_PATH"));
+  char *path = strtok(paths, ":");
+  while (path) {
+    strarray_push(&include_paths, path);
+    path = strtok(NULL, ":");
+  }

--- a/main.c
+++ b/main.c
@@ -0,5 +0,1 @@ static char *find_libpath(void) {
-  if (file_exists("/usr/lib/x86_64-linux-gnu/crti.o"))
-    return "/usr/lib/x86_64-linux-gnu";
-  if (file_exists("/usr/lib64/crti.o"))
-    return "/usr/lib64";
-  error("library path is not found");
+  return GLIBC_ROOT "/lib";

--- a/main.c
+++ b/main.c
@@ -0,5 +0,3 @@ static char *find_gcc_libpath(void) {
   char *paths[] = {
-    "/usr/lib/gcc/x86_64-linux-gnu/*/crtbegin.o",
-    "/usr/lib/gcc/x86_64-pc-linux-gnu/*/crtbegin.o", // For Gentoo
-    "/usr/lib/gcc/x86_64-redhat-linux/*/crtbegin.o", // For Fedora
+    GCC_ROOT "/lib/gcc/x86_64-unknown-linux-gnu/*/crtbegin.o",
   };

@@ -0,1 +0,1 @@ static void run_linker(StringArray *inputs, char *output) {
-  strarray_push(&arr, "ld");
+  strarray_push(&arr, LD_ROOT "/bin/ld");

--- a/main.c
+++ b/main.c
@@ -0,2 +0,2 @@ static void run_linker(StringArray *inputs, char *output) {
     strarray_push(&arr, "-dynamic-linker");
-    strarray_push(&arr, "/lib64/ld-linux-x86-64.so.2");
+    strarray_push(&arr, GLIBC_ROOT "/lib/ld-2.31.so");

--- a/main.c
+++ b/main.c
@@ -0,8 +0,7 @@ static void run_linker(StringArray *inputs, char *output) {
-  strarray_push(&arr, "-L/usr/lib/x86_64-linux-gnu");
-  strarray_push(&arr, "-L/usr/lib64");
-  strarray_push(&arr, "-L/lib64");
-  strarray_push(&arr, "-L/usr/lib/x86_64-linux-gnu");
-  strarray_push(&arr, "-L/usr/lib/x86_64-pc-linux-gnu");
-  strarray_push(&arr, "-L/usr/lib/x86_64-redhat-linux");
-  strarray_push(&arr, "-L/usr/lib");
-  strarray_push(&arr, "-L/lib");
+  strarray_push(&arr, format("-L%s", GCC_ROOT "/lib"));
+  char *paths = strdup(getenv("LIBRARY_PATH"));
+  char *path = strtok(paths, ":");
+  while (path) {
+    strarray_push(&arr, format("-L%s", path));
+    path = strtok(NULL, ":");
+  }
